app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java

```diff
@@ -0,0 +1,16 @@
+package ru.yandex.practicum.contacts.presentation.base;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.DiffUtil;
+
+public class BaseListDiffCallback<T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T> {
+    @Override
+    public boolean areItemsTheSame(@NonNull T oldItem, @NonNull T newItem) {
+        return oldItem.hashCode() == newItem.hashCode();
```

Comment:
```suggestion
        return oldItems.theSameAs(newItem);
```
Здесь нужно вызывать функцию из интерфейса `ListDiffInterface`

Reply:
Вопрос, ты написал в замечаниях следующее
- ⚠️ Критические замечания:
❌ Создан дженерик-класс BaseListDiffCallback с реализацией - Некорректная реализация метода areItemsTheSame
Метод я поправил, а вот с созданием дженерик-класса получается все, ошибка поправлена или нет?

Reply:
Ага, только в этом методе была ошибка

Reply:
@ArturNurtdinov Я отправил, не знал что если тот же пр, то заново ссылку подливать нужно.